api/desecapi/renderers.py

```diff
@@ -11,11 +12,23 @@ def render(self, data, media_type=None, renderer_context=None):
         response = renderer_context.get('response')
 
         if response and response.exception:
-            # TODO enable the renderer to handle responses where reponse.data is a list
-            if not isinstance(data, dict) or data.get('detail', None) is None:
-                raise ValueError('Expected response.data to be a dict with error details in response.data[\'detail\'], '
-                                 'but got %s:\n\n%s' % (type(response.data), response.data))
             response['Content-Type'] = 'text/plain'
-            return data['detail']
+
+            try:
+                return data['detail']
+            except KeyError:
+                pass
+
+            try:
+                details = [el['detail'] for el in data if getattr(el, 'detail', None) is not None]
```

Comment:
redundant, as the `'detail'` is looked up twice from `el` if `not None`. Also, `detail` is not an attribute of `el` (it is a dictionary key), so you probably mean `el.get('detail')` instead of `getattr`. How about:

`details = list(filter(None, [el.get('detail') for el in data]))`

See the note in https://docs.python.org/3/library/functions.html#filter

(PS: obfuscation contest: `list(filter(None, list(map(lambda x: x.get('detail'), data))))`)